### PR TITLE
feat: Add support for (React) JSX and TSX files

### PR DESCRIPTION
**Pull request by charles-prof**
This commit adds support for react tsx/jsx files and intellisense for className prop . This feature allows developers to have a better coding experience when working with Bootstrap utility classes in React components.

### CODE REVIEW

File: vscode-intellisense/src/completion/utils/isOffsetPossiblyInAClass.test.ts [hunk 1]
```diff
@@ -30,13 +30,13 @@ describe("isOffsetPossiblyInAClass", () => {
     test('it recognizes "class" attr', () => {
       const documentText = `<div class=""></div>`;
       const offset = 11;
-      expect(_isStartOfClassValue(documentText, offset)).toBe(true);
+      expect(_isStartOfClassorClassNameValue(documentText, offset)).toBe(true);
```
Comment (shaman-apprentice) [inline vscode-intellisense/src/completion/utils/isOffsetPossiblyInAClass.test.ts]: I think we should rename description of above `describe`-block to reflect the new function name and add a test case, that it recognizes "className" attr.

Would you mind renaming the function to `_isStartOfClassOrClassNameValue`, so it is completely camel case?

File: vscode-intellisense/src/completion/utils/isOffsetPossiblyInAClass.ts [hunk 2]
```diff
@@ -2,20 +2,16 @@ export function isOffsetPossiblyInAClass(text: string, offset: number): boolean
...
-  if (text[--offset]?.toLowerCase() !== "=") return false;
-  if (text[--offset]?.toLowerCase() !== "s") return false;
-  if (text[--offset]?.toLowerCase() !== "s") return false;
-  if (text[--offset]?.toLowerCase() !== "a") return false;
-  if (text[--offset]?.toLowerCase() !== "l") return false;
-  if (text[--offset]?.toLowerCase() !== "c") return false;
-  return true;
-}
+export const _isStartOfClassorClassNameValue = (text: string, offset: number) =>
+  text.slice(offset - 6, offset).toLowerCase() === "class=" || text.slice(offset - 10, offset) === "className=";
```
Comment (shaman-apprentice) [inline vscode-intellisense/src/completion/utils/isOffsetPossiblyInAClass.ts:14]: nice :)

File: vscode-intellisense/src/completion/utils/isOffsetPossiblyInAClass.ts [hunk 3]
```diff
@@ -2,20 +2,16 @@ export function isOffsetPossiblyInAClass(text: string, offset: number): boolean
...
+
+// export const _isStartOfClassNameValue = (text: string, offset: number) =>
```
Comment (shaman-apprentice) [inline vscode-intellisense/src/completion/utils/isOffsetPossiblyInAClass.ts]: Please remove the comment.

**Review by shaman-apprentice — CHANGES_REQUESTED** [review]
Thanks for adding the feature. Looks great. Just a few minor notes. If you have time to address them, that would be great. Otherwise, if you don't mind, I will merge your PR and address them afterwards.

File: vscode-intellisense/src/completion/utils/isOffsetPossiblyInAClass.ts (hunk 3 above)
Comment (charles-prof) [reply vscode-intellisense/src/completion/utils/isOffsetPossiblyInAClass.ts]: Hi, I augmented the test and improved the naming convention.

**Review by shaman-apprentice — APPROVED** [review]
Thanks! I am not sure, if I will find the time to release today. But tomorrow for sure.

**Comment by charles-prof** [comment]
No Mention ! Your extension is very useful....

**Pull request merged**